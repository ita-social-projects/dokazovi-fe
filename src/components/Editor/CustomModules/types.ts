import Quill from 'quill';
import { DropEvent, FileRejection } from 'react-dropzone';

export interface IUrlInputModalProps {
  editor?: Quill;
  updateBackgroundImage?: (url: string) => void;
  forBackgroundImg?: boolean;
}

export interface IBackgroundImageContainerProps {
  dispatchImageUrl: (backgroundImageUrl: string) => void;
  fileSelectorHandler: <T extends File>(
    acceptedFiles: T[],
    fileRejections: FileRejection[],
    event: DropEvent,
  ) => void;
  title?: string;
  imgUrl?: string;
  notCarousel?: boolean;
  reminder?: boolean;
  forMobilePic: boolean;
}

export interface IFileInputProps {
  onDrop: <T extends File>(
    acceptedFiles: T[],
    fileRejections: FileRejection[],
    event: DropEvent,
  ) => void;
  forMobilePic: boolean;
}
