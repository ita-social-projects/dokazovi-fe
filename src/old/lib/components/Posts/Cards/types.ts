import { IPost } from '../../../types';

export interface IPostPreviewCardProps {
  post: IPost;
  shouldNotUseLink?: boolean;
}
