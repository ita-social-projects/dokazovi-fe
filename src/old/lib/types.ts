export enum ExpertStatus {
  NEW = 'NEW',
  ACTIVE = 'ACTIVE',
  DELETED = 'DELETED',
}
export enum PostStatus {
  DRAFT = 'DRAFT',
  MODERATION_FIRST_SIGN = 'MODERATION_FIRST_SIGN',
  MODERATION_SECOND_SIGN = 'MODERATION_SECOND_SIGN',
  PUBLISHED = 'PUBLISHED',
  ARCHIVED = 'ARCHIVED',
}

export enum OriginTypeEnum {
  EXPERT_OPINION = 1,
  TRANSLATION = 2,
  MEDIA = 3,
}

export type OriginType =
  | OriginTypeEnum.EXPERT_OPINION
  | OriginTypeEnum.MEDIA
  | OriginTypeEnum.TRANSLATION;

export enum PostTypeEnum {
  ARTICLE = 1,
  DOPYS = 2,
  VIDEO = 3,
}

export enum LoadMoreButtonTextType {
  EXPERT,
  POST,
}

export enum PostsPreviewCardStylesEnum {
  VIDEO = 'VIDEO',
  MEDIA = 'MEDIA',
  TRANSLATION = 'TRANSLATION',
  EXPERT_OPINION = 'EXPERT_OPINION',
  IMPORTANT = 'IMPORTANT',
}

export type PostsPreviewCardStylesType =
  | PostsPreviewCardStylesEnum.VIDEO
  | PostsPreviewCardStylesEnum.MEDIA
  | PostsPreviewCardStylesEnum.TRANSLATION
  | PostsPreviewCardStylesEnum.EXPERT_OPINION
  | PostsPreviewCardStylesEnum.IMPORTANT;

export enum LoadingStatusEnum {
  idle = 'idle',
  pending = 'pending',
  succeeded = 'succeeded',
  failed = 'failed',
}

export type LoadingStatusType =
  | LoadingStatusEnum.idle
  | LoadingStatusEnum.pending
  | LoadingStatusEnum.succeeded
  | LoadingStatusEnum.failed;

export enum LocalStorageKeys {
  ACCESS_TOKEN = 'ACCESS_TOKEN',
}

export enum FilterTypeEnum {
  POST_TYPES,
  DIRECTIONS,
  REGIONS,
  TAGS,
  ORIGINS,
}

export enum ChipFilterEnum {
  REGION = 'REGION',
  DIRECTION = 'DIRECTION',
  ORIGIN = 'ORIGIN',
  POST_TYPE = 'POST_TYPE',
}

export type ChipFilterType =
  | ChipFilterEnum.REGION
  | ChipFilterEnum.DIRECTION
  | ChipFilterEnum.ORIGIN
  | ChipFilterEnum.POST_TYPE;

export enum QueryTypeEnum {
  POST_TYPES = 'types',
  DIRECTIONS = 'directions',
  REGIONS = 'regions',
  TAGS = 'tags',
  PAGE = 'page',
  ORIGINS = 'origins',
}

export interface IPost {
  id: number;
  title: string;
  content: string;
  author: {
    avatar?: string;
    firstName: string;
    id: number;
    lastName: string;
    mainInstitution: {
      city: {
        id: number;
        name: string;
      };
      id: number;
      name: string;
    };
  };
  directions: IDirection[];
  origin: IOrigin[];
  tags?: IPostTag[];
  type: IPostType;
  createdAt: string;
  modifiedAt?: string;
  origins?: IOrigin[];
  preview: string;
  previewImageUrl?: string;
  videoUrl?: string;
}

export interface IExpert {
  id: number;
  firstName: string;
  lastName: string;
  avatar?: string;
  qualification?: string;
  phone?: string;
  email?: string;
  bio?: string;
  mainInstitution?: IInstitution;
  mainDirection?: IDirection;
  directions?: IDirection[];
  lastAddedPost?: {
    id: number;
    title: string;
  };
}

export interface IPostTag {
  id: number;
  tag: string;
}

export interface IPostType {
  id: number;
  name: string;
}

export interface IDirection {
  id: number;
  color?: string;
  name: string;
  label?: string;
  hasPosts?: boolean;
}

export interface IFilter {
  id: number;
  color?: string;
  name: string;
  label?: string;
}

export interface IOrigin {
  id: number;
  name: string;
  parameter: null | string | number;
}

export interface IRegion {
  id: number;
  name: string;
  usersPresent?: boolean;
}

export interface IInstitution {
  city: {
    id: number;
    name: string;
  };
  id: number;
  name: string;
}

export interface ICheckboxes {
  [key: string]: boolean;
}

export interface IAuthInputs {
  email: string;
  password: string;
}
