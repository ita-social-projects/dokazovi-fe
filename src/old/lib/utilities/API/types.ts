import { IDirection, IOrigin, IPost } from '../../types';

export type CreatePostRequestType = {
  title: string;
  authorsName?: string;
  authorsDetails?: string;
  content: string;
  directions: IDirection[];
  origins?: IOrigin[];
  preview: string;
  type: {
    id: number;
  };
  previewImageUrl?: string;
  videoUrl?: string;
  authorId?: number | null;
};

export type CreateTextPostRequestType = CreatePostRequestType;

export type CreateVideoPostRequestType = CreatePostRequestType & {
  videoUrl: string;
};

export type CreatePostRequestUnionType =
  | CreateTextPostRequestType
  | CreateVideoPostRequestType;

type UpdatePostRequestType = CreatePostRequestType & {
  id: number;
};

export type UpdateTextPostRequestType = UpdatePostRequestType;

export type UpdateVideoPostRequestType = UpdatePostRequestType & {
  videoUrl: string;
};

export type UpdatePostRequestUnionType =
  | UpdateTextPostRequestType
  | UpdateVideoPostRequestType;

export type PostResponseType = {
  author: {
    avatar: string;
    firstName: string;
    id: number;
    lastName: string;
    mainInstitution: {
      city: {
        id: number;
        name: string;
      };
      id: number;
      name: string;
    };
  };
  content: string;
  preview: string;
  postType: PostTypeResponseType;
  createdAt: string;
  directions: DirectionResponseType[];
  origins: IOrigin[];
  id: number;
  title: string;
  type: {
    id: number;
    name: string;
  };
  videoUrl?: string;
};

export type ExpertResponseType = {
  avatar: string;
  firstName: string;
  id: number;
  lastAddedPost: {
    id: number;
    title: string;
  };
  lastName: string;
  mainDirection: {
    id: number;
    name: string;
  };
  mainInstitution: {
    city: {
      id: number;
      name: string;
    };
    id: number;
    name: string;
  };
  qualification: string;
};

export type CreateTagRequestType = {
  tag: string;
};

export type TagResponseType = {
  id: number;
  tag: string;
};

export type VersionResponseType = {
  version: string;
};

export type DirectionResponseType = {
  id: number;
  name: string;
  label: string;
  color: string;
};

export type OriginResponseType = {
  id: number;
  name: string;
  parameter: null;
};

export type RegionResponseType = {
  id: number;
  name: string;
};

export type PostTypeResponseType = {
  id: number;
  name: string;
};

export type LoginResponseType = {
  accessToken: string;
  tokenType: string;
};

type GetResponseType<T> = {
  content: T[];
  empty: boolean;
  first: boolean;
  last: boolean;
  number: number;
  numberOfElements: number;
  size: number;
  sort: {
    empty: boolean;
    sorted: boolean;
    unsorted: boolean;
  };
  totalElements: number;
  totalPages: number;
};

export type ExpertsResponseType = GetResponseType<ExpertResponseType>;

export type PostsResponseType = GetResponseType<PostResponseType>;

export type NewestPostsResponseType = GetResponseType<NewestPostResponseType>;

export enum NewestTypeEnum {
  MEDIA,
  TRANSLATION,
  EXPERT_OPINION,
  VIDEO,
}

export type NewestPostResponseType = {
  fieldName: string;
  postDTOS: IPost[];
};

export type GetTagsConfigType = {
  params: {
    value: string;
    limit?: number;
  };
};

type GetConfigType = {
  params: RequestParamsType;
};

export type GetExpertsConfigType = GetConfigType & {
  params: {
    directions?: number[];
    regions?: number[];
  };
};

export type GetPostsConfigType = GetConfigType;

export type RequestParamsType = {
  page?: number;
  size?: number;
  sort?: string[];
  directions?: number[];
  type?: number[];
  types?: number[];
  expert?: number;
  regions?: number[];
  tag?: number[];
  origins?: number[];
  userName?: string;
};
