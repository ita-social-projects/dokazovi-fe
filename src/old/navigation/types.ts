export interface IRouteConfig {
  component: React.ComponentType;
  path: string | string[];
  key: string;
  exact?: boolean;
  private?: boolean;
  useRender?: boolean;
}
