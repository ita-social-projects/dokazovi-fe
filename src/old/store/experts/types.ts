import { LoadingStatusEnum } from '../../lib/types';
import { IMaterials } from '../materials/types';

export interface IExpertPayload {
  expertIds: number[];
  meta: IExpertMeta;
}

export interface IExpertMeta {
  totalPages: number;
  pageNumber: number;
  loading: LoadingStatusEnum;
  error: null | string;
  isLastPage: boolean;
  totalElements: number;
}

export interface IExpertsState {
  experts: IExpertPayload;
  materials: IMaterials;
}

export interface IFetchExpertsOptions {
  page: number;
  regions: number[];
  directions: number[];
  appendExperts: boolean;
}
