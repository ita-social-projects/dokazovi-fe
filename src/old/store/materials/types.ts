import { QueryTypeEnum, LoadingStatusEnum, IPost } from '../../lib/types';

export interface IMaterials {
  postIds: number[];
  meta: IMaterialsMeta;
} // Old type, used in expertsSlice, it will be removed;

interface IData {
  postIds: number[];
  meta: IMaterialsMeta;
  posts: {
    [id: string]: IPost;
  };
}

export interface IMaterialsState {
  data: IData;
  loading: LoadingStatusEnum;
  error: null | string;
  filters?: {
    [key in QueryTypeEnum]?: number[];
  };
}

export interface IFetchMaterialsOptions {
  filters: {
    page: number;
    postTypes: number[];
    directions: number[];
  };
  page: number;
  appendPosts: boolean;
}

export interface IMaterialsMeta {
  loading: LoadingStatusEnum;
  error: null | string;
  isLastPage: boolean;
  pageNumber: number;
  totalPages: number;
  totalElements: number;
} // Properties loading & error are used in expertsSlice, they will be removed from this interface;
