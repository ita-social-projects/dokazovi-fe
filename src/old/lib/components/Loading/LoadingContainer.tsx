import React from 'react';
import { Grid } from '@material-ui/core';
import { LoadingStatusEnum } from '../../types';
import { useStyles } from './LoadingContainer.styles';
import { LoadingInfo } from './LoadingInfo';

export interface ILoadingContainerProps {
  loading: LoadingStatusEnum;
  errorMsg?: string;
  expand?: boolean;
}

export const LoadingContainer: React.FC<ILoadingContainerProps> = (props) => {
  const { loading, errorMsg, expand } = props;
  const classes = useStyles();

  return (
    <Grid
      container
      className={expand ? classes.expandedContainer : classes.container}
    >
      <LoadingInfo loading={loading} errorMsg={errorMsg} />
    </Grid>
  );
};
