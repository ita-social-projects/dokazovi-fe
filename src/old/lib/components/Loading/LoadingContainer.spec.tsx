import React from 'react';
import { render } from '@testing-library/react';
import { ILoadingContainerProps, LoadingContainer } from './LoadingContainer';
import { LoadingStatusEnum } from '../../types';

const PROPS_MOCK: ILoadingContainerProps = {
  loading: LoadingStatusEnum.idle,
  expand: false,
};

describe('LoadingContainer', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <LoadingContainer loading={PROPS_MOCK.loading} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
