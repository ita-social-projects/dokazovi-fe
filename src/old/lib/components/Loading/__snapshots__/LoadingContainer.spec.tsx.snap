// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingContainer should render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root LoadingContainer-container-2 MuiGrid-container"
  />
</DocumentFragment>
`;
